src/server/utils/devCdnFactory.js

```diff
@@ -0,0 +1,194 @@
+import path from 'path';
+import fs from 'fs';
+import got from 'got';
+import cors from '@fastify/cors';
+import compress from '@fastify/compress';
+import fastifyStatic from '@fastify/static';
+import Fastify from 'fastify';
+import ip from 'ip';
+import ProxyAgent from 'proxy-agent';
+
+const getLocalModuleMap = ({ pathToModuleMap, oneAppDevCdnAddress }) => {
+  const moduleMap = JSON.parse(fs.readFileSync(pathToModuleMap, 'utf8').toString());
+  Object.keys(moduleMap.modules).forEach((moduleName) => {
+    const module = moduleMap.modules[moduleName];
+    module.browser.url = module.browser.url.replace('[one-app-dev-cdn-url]', oneAppDevCdnAddress);
+    module.legacyBrowser.url = module.legacyBrowser.url.replace('[one-app-dev-cdn-url]', oneAppDevCdnAddress);
+    module.node.url = module.node.url.replace('[one-app-dev-cdn-url]', oneAppDevCdnAddress);
+  });
+  return JSON.stringify(moduleMap, null, 2);
+};
+
+const matchPathToKnownRemoteModuleUrl = (
+  incomingRequestPath,
+  remoteModuleBaseUrls
+) => remoteModuleBaseUrls.find((remoteModuleBaseUrl) => {
+  const remoteModuleUrlOrigin = new URL(remoteModuleBaseUrl).origin;
+  const remoteModulePath = remoteModuleBaseUrl.replace(remoteModuleUrlOrigin, '');
+
+  return incomingRequestPath.startsWith(remoteModulePath);
+});
+
+const consumeRemoteRequest = async (remoteModuleMapUrl, hostAddress, remoteModuleBaseUrls) => {
+  try {
+    const response = await got(remoteModuleMapUrl, {
+      agent: {
+        https: new ProxyAgent(),
+        http: new ProxyAgent(),
+      },
+    });
+
+    // clear out remoteModuleBaseUrls as the new module map now has different urls in it
+    // not clearing would result in an ever growing array
+    remoteModuleBaseUrls.splice(0, remoteModuleBaseUrls.length);
```

Comment:
as a preemptive explanation of this line, I had to clear the array in place as if I just re-assigned it like so 
```js remoteModuleBaseUrls = []``` it does not modify the original array which needs to be done